server/src/main/java/org/opensearch/search/SearchBootstrapSettings.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search;
+
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+
+/**
+ * Keeps track of all the search related node level settings which can be accessed via static methods
+ */
+public class SearchBootstrapSettings {
+    // settings to configure maximum slice created per search request using OS custom slice computation mechanism. Default lucene
```

Comment:
Is there any question as to whether we will switch away from the static method to the dynamic method when the next release of Lucene is available? If not, I'd go ahead and create a GitHub issue to track it and link the issue in a comment in the code where appropriate.

Reply:
@sohami to this point, why this setting are static? As far as I can tell, it is used in non-static context and could be implemented as a regular search-related setting.

Reply:
thats interesting, do we usually update `main` with the lucene snapshot builds as well ? Was wondering if we take any dependency on a new change in unreleased lucene version and that gets changed in released lucene version then the change will break. 

But nonetheless we can keep this change as is and backport to 2.x until 9.8 is officially released. Then add a follow-up commit to move it to dynamic setting as part of https://github.com/opensearch-project/OpenSearch/issues/8870

Reply:
> thats interesting, do we usually update main with the lucene snapshot builds as well ?

Always

> Was wondering if we take any dependency on a new change in unreleased lucene version and that gets changed in released lucene version then the change will break.

That is why we update to latest snapshots regularly 

> But nonetheless we can keep this change as is and backport to 2.x until 9.8 is officially released. Then add a follow-up commit to move it to dynamic setting as part of https://github.com/opensearch-project/OpenSearch/issues/8870

I don't think we should go with statics here - it is very likely that 9.8.0 will be released before 2.10 so we could properly integrate the change.

Reply:
> But nonetheless we can keep this change as is and backport to 2.x until 9.8 is officially released. Then add a follow-up commit to move it to dynamic setting as part of https://github.com/opensearch-project/OpenSearch/issues/8870

>> I don't think we should go with statics here - it is very likely that 9.8.0 will be released before 2.10 so we could properly integrate the change.

If the 9.8.0 is released in 2.10 timeframe then we will change this to use the dynamic setting. If not we still want to be able to use this mechanism to do all the performance benchmark. This is done because we are not sure if 9.8.0 will be released by 2.10 and have some static mechanism meanwhile.

Reply:
@sohami you did the change specifically to support this feature, and we have the opportunity to use it right away, why not? If you really want to have it in 2.x, we could use this pull request against 2.x branch (as the temporary measure) and have a follow up issue to migrate to the proper integration with 9.8.0 release (we have done that a few times). 

> This is done because we are not sure if 9.8.0 will be released by 2.10 and have some static mechanism meanwhile.

Why gathering statistics from 3.0.0 is not sufficient?

Reply:
> If you really want to have it in 2.x, we could use this pull request against 2.x branch (as the temporary measure) and have a follow up issue to migrate to the proper integration with 9.8.0 release (we have done that a few times).

Yes I want to have this support in 2.x as well. With that I was thinking if we make this change in `main` and backport to `2.x` then there won't be any custom changes in `2.x`. Since now `main` is already moved to 9.8.0-snapshot version of lucene, I will add a follow-up PR to clean this up in main and have a dynamic mechanism but not backport that to 2.x. When 9.8.0 is released either in 2.10 or future 2.x versions we can backport the new PR to that version which would probably we a cleaner port as it will be built on top of this change ? I have a tracking follow-up issue created for this dynamic change (https://github.com/opensearch-project/OpenSearch/issues/8870). If we make this change specifically in 2.x and new change in main, then will need to ensure proper cleanup with 2 different PRs which can be messy ?

> Why gathering statistics from 3.0.0 is not sufficient?

We are targeting it for 2.10 and don't want to take changes in 3.x which may not be part of that release.

Reply:
> Since now main is already moved to 9.8.0-snapshot version of lucene, I will add a follow-up PR to clean this up in main and have a dynamic mechanism but not backport that to 2.x. 

I am trying to understand why do you want to get this change in `main`? We know this is not the way to go forward, and we do have the solution, you will have to redo this work in two branches instead of just cherry picking one simple change into 2.x.

Reply:
>  When me move this new PR (pr_2) from main to 2.x, we will need to ensure the cleanup like for SearchBootstrapSettings and other conflicts happens properly and not get missed (which I was thinking can be messy as these 2 PRs are not built on top of each other).

This is inevitable in any case - we will be backporting all changes related to Lucene 9.8.0 (as we did for all previous Apache Lucene versions), the argument here is: keep main clean by using the new Apache Lucene snapshots (this is why we've been always doing that), do backport as a best effort.

Reply:
I will close this PR and open a new one against 2.x. Will make the dynamic setting change as separate PR for main branch. And will create another tracking backport issue for merging the dynamic setting change to 2.x when lucene 9.8 gets released. As part of that backport we can revert the commit for static setting in 2.x and then apply the commit from main.

Reply:
@reta @andrross @jed326 Created PR https://github.com/opensearch-project/OpenSearch/pull/8884 for 2.x branch. Will be working on dynamic setting change for main branch.